tests/testthat/test-multi-file.R

```diff
@@ -111,6 +111,29 @@ test_that("vroom works if a file contains no data", {
   expect_equal(res, tibble::tibble(A = 1, B = 2))
 })
 
+test_that("vroom works if some files contain no data, regardless of order (#430)", {
```

Comment:
I would love to see these files have self-explanatory names, like "header-but-no-data.csv" or "header-and-one-data-row.csv".

I realize this might force you to tweak some other tests that use these same files, but it makes the intent of each test so much more clear.

Reply:
Alternatively, if you don't want to touch any other tests, create the exact files you need for this, with good names. These are tiny files, so there's no need to worry about adding files of this size. If/when we touch other tests that could use these new, self-documenting test files, we can switch them over.

And actually: I would probably just inline the creation (and destruction) of these test files. The combination of explicit construction and self-explaining file names would make the intent of each expectation crystal clear.

In this case, I would use `withr::local_tempfile()` to create the test files.